### DISCUSSION TITLE
Regression (change?) from 0.6: POST with status code doesn't compile

### DISCUSSION THREAD

**Discussion by moh-eulith** [tokio-rs/axum#2662] [Q&A] [answered]
- [ x ] I have looked for existing issues (including closed) about this
searched for post and handler 

### Version
0.7.4

```
│   │   ├── axum v0.6.20
│   │   │   ├── axum-core v0.3.4
│   ├── axum v0.6.20 (*)
├── axum v0.7.4
│   ├── axum-core v0.4.3
│   ├── axum v0.7.4 (*)
│   ├── axum v0.7.4 (*)
```
The 0.6 versions are a deep dependency.
### Platform
Linux

### Description
```rust
Router::new().route(
        "/json",
        post(|payload: Json<serde_json::Value>| async move {
            (StatusCode::OK, payload.to_string())
        })
```
no longer compiles in 0.7.4. Used to compile in 0.6. the problem is the status code. just `payload.to_string()`, for example, compiles.
Not sure if this is a regression or an intended change. Nothing in the changelogs...

**Comment by jplatte** [comment, marked as answer]
Hi, could it be that you're trying to return an `http 0.2.x` status code? Try importing from `axum::http`, or upgrading the crate if so.

**Reply by moh-eulith** [reply]
You're correct, it's an http crate version issue. I wasn't trying to return that particular version, it was being pulled in via another crate (reqwest).  Dealing with it now. Please close. Thanks and sorry for the noise.
